File: 品味Java的21種設計模式/dp_java/flyweight/example1/AuthorizationModel.java
package cn.javass.dp.flyweight.example1;
/**
 * 描述授权数据的数据model
 */
public class AuthorizationModel {
	/**
	 * 人员
	 */
	private String user;
	/**
	 * 安全实体
	 */
	private String securityEntity;
	/**
	 * 权限
	 */
	private String permit;
	
	public String getUser() {
		return user;
	}
	public void setUser(String user) {
		this.user = user;
	}
	public String getSecurityEntity() {
		return securityEntity;
	}
	public void setSecurityEntity(String securityEntity) {
		this.securityEntity = securityEntity;
	}
	public String getPermit() {
		return permit;
	}
	public void setPermit(String permit) {
		this.permit = permit;
	}
	
}

File: 品味Java的21種設計模式/dp_cpp/flyweight/example1/AuthorizationModel.cpp
#include "AuthorizationModel.h"

namespace cn {
namespace javass {
namespace dp {
namespace flyweight {
namespace example1 {

std::string AuthorizationModel::getUser() {
    return user;
}

void AuthorizationModel::setUser(std::string user) {
    this->user = user;
}

std::string AuthorizationModel::getSecurityEntity() {
    return securityEntity;
}

void AuthorizationModel::setSecurityEntity(std::string securityEntity) {
    this->securityEntity = securityEntity;
}

std::string AuthorizationModel::getPermit() {
    return permit;
}

void AuthorizationModel::setPermit(std::string permit) {
    this->permit = permit;
}
}
}
}
}
}
